Tests/MongoSwiftTests/MongoCollectionTests.swift

```diff
@@ -110,6 +111,11 @@ final class MongoCollectionTests: XCTestCase {
             .to(equal([["cat": "dog"], ["cat": "cat"]] as [Document]))
     }
 
+    func testBulkWrite() throws {
+        // TODO: check for MongoError.invalidArgument
+        expect(try self.coll.bulkWrite([])).to(throwError())
```

Comment:
@kmahar: The examples in [Nimble: Swift Error Handling](https://github.com/Quick/Nimble#swift-error-handling) didn't seem useful to assert the type of error thrown (I hoped that one of the two final examples would have worked). I also didn't see any prior art in the driver tests for doing so.

Do you have any idea how we can check for this?

Reply:
Ah, like to check for a particular enum case? I think we could do something like
```swift
do {
    try self.coll.bulkWrite([])
    // trigger a failure if we actually make it to this line
} catch {
   // cast error to a MongoError and fail if it's not the .invalidArgument case
}
```

I'm not sure how much work it would be, but I would be in favor of adding a custom Nimble matcher that executes a closure and checks that a particular `MongoError` case is thrown. Something like
`expect(try self.coll.bulkWrite([])).to(throwMongoError(.invalidArgument))`.